compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -2863,7 +2830,7 @@ private void generateMappingAccess(BLangIndexBasedAccess astIndexBasedAccessExpr
             } else if (types.isAssignable(astAccessExprExprType, symTable.xmlType)) {
                 generateXMLAccess((BLangXMLAccessExpr) astIndexBasedAccessExpr, tempVarRef, varRefRegIndex,
                         keyRegIndex);
-                this.varAssignment = variableStore;
```

Comment:
Shall we use this.varAssignment directly in the if condition in the previous check?

Reply:
Same here as well